reflekt-plugin/src/main/kotlin/io/reflekt/plugin/analysis/processor/Util.kt

```diff
@@ -10,14 +12,67 @@ internal val KtElement.isPublicObject: Boolean
 internal val KtElement.isPublicFunction: Boolean
     get() = this is KtNamedFunction && this.isPublic
 
+internal val KtElement.isTopLevelPublicFunction: Boolean
+    get() = isPublicFunction && this is KtNamedFunction && this.isTopLevel
```

Comment:
redundant check for `KtNamedFunction` as is already checked in `isPublicFunction`